### PR TITLE
Allow for no default in keywords

### LINKED ISSUE #6: Need default value to function?
It is observed in https://github.com/Moelf/FHist.jl/pull/83 that you need default value for the keyword argument for this package to work, is this intended?

### PR DESCRIPTION
**Pull request by MilesCranmer**
Fix #6

@moelf could you please review?

### CODE REVIEW

File: src/DeprecateKeywords.jl [hunk 1]
```diff
@@ -65,10 +65,12 @@ function _depkws(def)
 
     # Update new symbols to use deprecated kws if passed:
     for (i, kw) in enumerate(sdef[:kwargs])
-        isa(kw, Symbol) && continue
-
-        new_kw = kw.args[1]
-        default = kw.args[2]
+        no_default = isa(kw, Symbol)
```
Comment (MilesCranmer) [inline src/DeprecateKeywords.jl:68]: Actually this does not account for things like `; a::Int`. Needs to be more general.

Comment (MilesCranmer) [reply src/DeprecateKeywords.jl:68]: Fixed.

**Comment by MilesCranmer** [comment]
Okay now you can also have type assertions. For example:

```julia
@depkws f(; (@deprecate a b), b::Int) = b
```

expands to

```julia
:(function f(; a = DeprecateKeywords.DeprecatedDefault, b = begin
                  #= /Users/mcranmer/Documents/DeprecateKeyword.jl/src/DeprecateKeywords.jl:94 =#
                  if a !== DeprecateKeywords.DeprecatedDefault
                      #= /Users/mcranmer/Documents/DeprecateKeyword.jl/src/DeprecateKeywords.jl:95 =#
                      Base.depwarn("Keyword argument `a` is deprecated. Use `b` instead.", :f)
                      #= /Users/mcranmer/Documents/DeprecateKeyword.jl/src/DeprecateKeywords.jl:96 =#
                      a
                  else
                      #= /Users/mcranmer/Documents/DeprecateKeyword.jl/src/DeprecateKeywords.jl:98 =#
                      DeprecateKeywords.DeprecatedDefault
                  end
              end)
      if b === DeprecateKeywords.DeprecatedDefault
          throw(UndefKeywordError(:b))
      end
      b::Int
      #= REPL[2]:1 =#
      b
  end)
```

**Comment by bojohnson5** [comment]
@MilesCranmer Sorry it's taken me a while to get to this. I'm using this branch in my testing and it looks like it makes it so both the deprecated keyword and new keyword are possible to use, with no deprecated keyword warning when using the old one.

As an example:
```
julia> Hist1D(Float64, bins=-1:1)
edges: -1:1
bin counts: [0.0, 0.0]
total count: 0.0
```

```
julia> Hist1D(Float64, binedges=-1:1)
edges: -1:1
bin counts: [0.0, 0.0]
total count: 0.0
```

while
```
julia> Hist1D(Float64, binedge=-1:1)
ERROR: MethodError: no method matching Hist1D(::Type{Float64}; binedge::UnitRange{Int64})

Closest candidates are:
  Hist1D(::Type{T}; bins, binedges, overflow) where T got unsupported keyword argument "binedge"
   @ FHist none:0
  Hist1D(::Any, ::StatsBase.AbstractWeights, ::AbstractRange; overflow) got unsupported keyword argument "binedge"
   @ FHist ~/.julia/dev/FHist/src/hist1d.jl:49
  Hist1D(::Any, ::StatsBase.AbstractWeights, ::AbstractVector) got unsupported keyword argument "binedge"
   @ FHist ~/.julia/dev/FHist/src/hist1d.jl:56
  ...

Stacktrace:
 [1] kwerr(::NamedTuple{(:binedge,), Tuple{UnitRange{Int64}}}, ::Type, ::Type)
   @ Base ./error.jl:165
 [2] Hist1D
   @ ~/.julia/packages/DeprecateKeywords/3IaTF/src/DeprecateKeywords.jl:98 [inlined]
 [3] top-level scope
   @ REPL[12]:1
```

**Comment by MilesCranmer** [comment]
> both the deprecated keyword and new keyword are possible to use

This is the same for keywords without defaults. Sadly I don’t think there’s a way to avoid it as it would break generics.

> with no deprecated keyword warning when using the old one

Cannot reproduce. Did you forget to start Julia with `--depwarn=yes`?

**Comment by bojohnson5** [comment]
That's what I was missing. I can confirm I get the correct behavior now when starting Julia with `--depwarn=yes`

**Comment by MilesCranmer** [comment]
Awesome, thanks for checking!

**Pull request merged**